ADD: codetour로 문서 만들기

sunghyunl22 opened the pull request:
- VScode 플러그인 codetour 를 사용해서 레포지토리 자습서를 만든 이야기입니다.
- 맞춤법 검사 완료

Commit:
ADD: codetour로 문서 만들기

- VScode 플러그인 codetour 를 사용해서 자습서를 만든 이야기입니다.

goonookim commented on _posts/2021-02-28-become-streamer.md:
```diff
@@ -0,0 +1,132 @@
+
+위치를 예로 들면 가장 기본은 매체가 스크립트를 설치한 위치에 위젯을 설치하는 방식이고, 설정에 따라서 본문 텍스트의 가운데 위치를 자동으로 찾아 들어가 노출하거나 스크롤을 내릴 때는 안 보였다가 다시 올라오는 순간에 보인다거나 하는 등 자동 옵션들도 많습니다.
+
+![%E1%84%86%E1%85%AE%E1%86%AB%E1%84%89%E1%85%A5%E1%84%85%E1%85%B3%E1%86%AF%20%E1%84%86%E1%85%A1%E1%86%AB%E1%84%83%E1%85%B3%E1%86%AF%E1%84%83%E1%85%A1%E1%84%80%E1%85%A1%20%E1%84%89%E1%85%B3%E1%84%90%E1%85%B3%E1%84%85%E1%85%B5%E1%84%86%E1%85%A5%E1%84%80%E1%85%A1%20%E1%84%83%E1%85%AC%E1%86%AB%20%E1%84%8B%E1%85%B5%E1%84%8B%E1%85%A3%E1%84%80%E1%85%B5%20530e0c1a62ee4835a214b1df1fd627b0/Untitled.png](/t ...
```
문서의 대체 텍스트를 전반적으로 손보면 좋을 것 같습니다!

sunghyunl22 replied:
수정되었습니다!

goonookim commented on _posts/2021-02-28-become-streamer.md:
```diff
@@ -0,0 +1,132 @@
+tags: [ 이성현, codetour, vscode, 문서화 ]
+---
+
+# 문서를 만들다가 스트리머가 된 이야기
```
이 heading은 h1, 나머지는 h3인데요. 제 생각엔 모두 동일한 수준의 heading 같은데요. 문서 제목이 h1으로 제공되니 1 depth heading은 h2로 통일해서 제공하면 어떨까요?

sunghyunl22 replied:
넵 그렇게 수정하겠습니다. h3 인건 텍스트 크기를 조절하려고 임의로 넣은 것이었습니다.

sunghyunl22 replied:
수정되었습니다.

goonookim replied:
음 여전히 이 헤딩은 h1이네요. h1, h2, h3는 텍스트 크기라는 심미적인 목적 보다는 문서의 아웃라인 구조를 결정하는 의미적인 접근이 우선적으로 필요하다고 생각해요.

현재 구조는 아래와 같은데요.

 * (h1) Visual Studio Code의 Codetour 플러그인으로 문서 만들기
 * (h1) 문서를 만들다가 스트리머가 된 이야기 (글 제목과 동일한 h1)
   * (h2) 필요성 (문서를... 의 하위 단락)
     * (h3) 데이블 위젯의 복잡한 기능 (필요성의 하위 단락)

네 단락의 아웃라인이 의미적으로 보았을 때 의도하신 바와 맞을까요? 제 기준으로는 "문서를 만들다가 스트리머가 된 이야기"라는 단락은 독립 도입 단락으로 보이고, "필요성", "데이블 위젯의 복잡한 기능"은 같은 깊이의 나열식 단락이라고 이해가 됩니다. 그래서 제 이해를 기준으로 정리해보면 아래와 같은 아웃라인 구조가 적합해보여요.

 * (h1) Visual Studio Code의 Codetour 플러그인으로 문서 만들기
   * (h2) 문서를 만들다가 스트리머가 된 이야기
   * (h2) 필요성
   * (h2) 데이블 위젯의 복잡한 기능

sunghyunl22 replied:
@goonookim jekyll 블로그의 title 필드에 적은 제목이 h1 인것을 몰랐어요. heading 구조를 재조정 했습니다. depth 는 의도한 것이었긴 한데 다른 사람이 이해한 내용 기준으로 다시 묶는것이 좋아 보여서 군우님 제안을 토대로 약간 바꿔서 적용해봤습니다. 
상세히 설명해주셔서 감사합니다. 

- h1) Visual Studio Code의 Codetour 플러그인으로 문서 만들기
   - h2) 문서를 만들다가 스트리머가 된 이야기 -> 시작하며 (제목 변경)
   - h2) 필요성 -> 배경 (제목 변경)
      - h3) _데이블 위젯의 복잡한 기능_ -> 히스토리를 찾는데 시간이 많이 걸림 (제목 변경)
      - h3) _너무 큰 저장소_ -> 저장소 크기와 의존성 (제목 변경)
   - h2) 코드를 보며 설명하는 문서를 만들자
   - h2) 자습서의 특징
   - h2) codetour plugin 
   - ... 쭉 h2

goonookim commented on _posts/2021-02-28-become-streamer.md:
```diff
@@ -0,0 +1,132 @@
+
+저는 최민지 님과 함께 위젯 api 서버 및 유저 플러그인 스크립트를 담당하고 있습니다. 
+
+이 포스트에서는 팀 내에서 2020.1월 진행한 "담당 영역 전문성 향상" 기간 동안 VScode 플러그인 codetour 를 사용해서 코드와 함께 볼 수 있는 히스토리 위주의 설명문서를 만든 경험, 그리고 그 내용을 팀에 공유한 방법에 관해 이야기해보려 합니다.
```
2020.1월에서 2020 -> 2021 오타와 함께 "2021년 1월"과 같이 변경하면 어떨까요?

sunghyunl22 replied:
1년전 이야기를 적을 뻔 했네요. 말씀해 주신대로 수정했습니다. ㅎㅎ

goonookim commented on _posts/2021-02-28-become-streamer.md:
```diff
@@ -0,0 +1,132 @@
+
+### 전체 흐름을 파악하기엔 너무 큰 저장소
+
+위젯 API 서버와 유저 플러그인 코드는 서로 메세지를 주고받기 때문에 별개 레포로 나눠지지 않고 한 레포에 모여있습니다.  2021.2.27 기준 레포의 총 코드 라인 수를 보면, test/빌드/외부 라이브러리/문서 등을 제외한 순수 로직 코드만 세었을 때 29,000라인입니다. 여기에 infra 설정과 CI 서버의 빌드 코드, 패키지로 불러와 사용하는 데이블의 공용 모듈 코드도 로직 일부로 보면 레포 하나를 이해하기 위해 봐야 하는 코드만 3만 라인 정도가 됩니다. 저장소를 처음 보는 사람이라면 원하는 부분을 찾는데 시간이 오래 걸릴 수 있죠. 
```
2021.2.27 날짜 포맷을 "2021년 2월 27일" 같이 수정하면 어떨까요?

sunghyunl22 replied:
넵. 수정하였습니다.

goonookim commented on _posts/2021-02-28-become-streamer.md:
```diff
@@ -0,0 +1,132 @@
+
+### 전체 흐름을 파악하기엔 너무 큰 저장소
+
+위젯 API 서버와 유저 플러그인 코드는 서로 메세지를 주고받기 때문에 별개 레포로 나눠지지 않고 한 레포에 모여있습니다.  2021.2.27 기준 레포의 총 코드 라인 수를 보면, test/빌드/외부 라이브러리/문서 등을 제외한 순수 로직 코드만 세었을 때 29,000라인입니다. 여기에 infra 설정과 CI 서버의 빌드 코드, 패키지로 불러와 사용하는 데이블의 공용 모듈 코드도 로직 일부로 보면 레포 하나를 이해하기 위해 봐야 하는 코드만 3만 라인 정도가 됩니다. 저장소를 처음 보는 사람이라면 원하는 부분을 찾는데 시간이 오래 걸릴 수 있죠. 
```
"레포"라는 단어는 "Repository"의 구어적인 느낌이라 직관적으로 이해할 수 없을지도 모르겠다는 생각이 드네요. "저장소" 같이 수정하면 어떨까요?

sunghyunl22 replied:
레포, repository 등의 단어를 저장소로 모두 수정했습니다.

goonookim commented on _posts/2021-02-28-become-streamer.md:
```diff
@@ -0,0 +1,132 @@
+
+자습서의 특징
+
+1) 코드를 보며 설명한다.
```
markdown의 ordered list 를 쓰면 어떨까요?

sunghyunl22 replied:
1) 도 되는것처럼 보였는데 표준 문법을 보니 1. 처럼 써야했네요.  ㅠㅠ 수정했습니다.

goonookim commented on _posts/2021-02-28-become-streamer.md:
```diff
@@ -0,0 +1,132 @@
+- 문서에 코드위치를 지정할 수 있습니다.
+- 순서가 있어서 버튼으로 앞뒤를 오가며 문서를 볼 수 있습니다.
+
+[https://marketplace.visualstudio.com/items?itemName=vsls-contrib.codetour](https://marketplace.visualstudio.com/items?itemName=vsls-contrib.codetour)
```
링크 텍스트를 URL 대신 문서의 이름 등으로 제공하면 좋을 것 같습니다.

sunghyunl22 replied:
수정했습니다.

goonookim commented on _posts/2021-02-28-become-streamer.md:
```diff
@@ -0,0 +1,132 @@
+
+# 문서를 만들다가 스트리머가 된 이야기
+
+안녕하세요 데이블 Publisher Platform 팀의 이성현입니다. 
+
+저는 최민지 님과 함께 위젯 api 서버 및 유저 플러그인 스크립트를 담당하고 있습니다. 
```
같은 소개 목적이니 한 문단으로 묶어도 좋을 것 같네요.

sunghyunl22 replied:
수정되었습니다. 이 외에도 문단을 묶을 수 있는 글을 찾아 합쳤습니다.

goonookim reviewed (commented):
👏👏👏

Jongman-Seo commented on _posts/2021-02-28-become-streamer.md:
```diff
@@ -0,0 +1,132 @@
+
+사실 이해하기 어려운 레포를 쉬운 레포로 만들기 위해서는 리팩토링 + 기능 분리가 가장 좋다고 생각합니다. 코드가 문서일 정도로 깨끗한 상황이 이상적이라고 생각합니다. 하지만 본인도 아직 전체를 파악하지 못했기 때문에 구조 개선을 생각해보기 어려웠고 최소 3개월이 걸릴 큰 작업이라 언제 시작될지 불투명했습니다. 저는 일단 정리 겸 현 상황이라도 잘 이해할 수 있는 문서 준비를 시작했습니다. 
+
+혼자 코드를 열심히 보던 때를 떠올려보면 코드 동작이 어렵다기 보다는, 이 코드가 왜 존재하고 이 기능이 어디서 사용되는지를 몰라 막히는 경우가 많았기 때문에 코드 바로 옆에서 배경지식을 읽을 수 있어야 한다는 조건을 붙였습니다. 코드가 많으면 진행과정을 순서대로 이해하는것도 부담이기 때문에 문서는 기능단위로 자르지 않고 실제 기능이 동작하는 진행 순서를 기준으로 만들어 따로 정리하는 시간을 줄였습니다. 문서 outdate 문제에 대해서는 로직보다 진행과정과 배경지식, 사업의도를 중심으로 설명하면 리팩토링 등으로 코드가 변하더라도 업데이트하기 수월할것 같았습니다. 일단 전체를 코드레벨로 살펴보는 문서가 없었기 때문에 outdate 문제는 좀 미뤄둡니다.
```
맞춤법 검사기가 readme에 나온 것은 글자 수 제한이 있어 page를 계속 넘겨봐야 되는 것으로 알고 있습니다.
뒷쪽의 맞춤법 검사가 많이 누락된 느낌이네요.

https://www.saramin.co.kr/zf_user/tools/character-counter
요걸 써보심이 어떨지..

sunghyunl22 replied:
감사합니다. 제안해 주신 맞춤법 검사기로 다시 검사해 반영했습니다.

조민지님 말씀대로 검사를 할 때마다 내용이 달라져서 여러번 돌려 확인해야 하는건 좀 불편하네요.

Commit:
REVIEW: heading depth 조정

- title = h1 이어서 전체적으로 구조 조정
- 오타 수정
- jekyll cache 폴더 gitignore추가

https://github.com/teamdable/techblog/pull/45#discussion_r586433867

goonookim reviewed (approved):
👏 👏

sunghyunl22 commented:
@Jongman-Seo 시간되실 때 업데이트 판 체크한번 부탁드려요 😬
